### ISSUE TITLE
example crashes at # 2. Run ensemble lake models

### ISSUE THREAD

**Issue by rotifer16** [aemon-j/LakeEnsemblR#232]
Hi, 
the provided example crashes at # 2. Run ensemble lake models
installation worked smoothly
here my output

run_ensemble(config_file = config_file, model = model)
Loading water temperature observations...[2021-08-04 14:54:27]
Finished loading water temperature observations![2021-08-04 14:54:27]
Running models... (Have you tried parallelizing?) [2021-08-04 14:54:27]
FLake run is complete! [2021-08-04 14:54:27]
GLM run is complete! [2021-08-04 14:54:27]
sh: line 1:  4037 Abort trap: 6           DYLD_LIBRARY_PATH=/Library/Frameworks/R.framework/Versions/4.0/Resources/library/GOTMr/exec '/Library/Frameworks/R.framework/Versions/4.0/Resources/library/GOTMr/exec/macgotm' > /dev/null 2> /dev/null
GOTM run is complete! [2021-08-04 14:54:28]
Error in R_nc4_open: No such file or directory
Error in nc_open(ncdf) : 
  Error in nc_open trying to open file ./GOTM/output/output.nc

what can I do?
I had a look at the folder GOTM/output/output.nc and there was no file output.nc but I did not get any error message for the line # 1. export_config(config_file = config_file, model = model)

all the best,
Ulrike

**Comment by tadhg-moore** [comment]
Hi Ulrike,

I believe @robertladwig has been in touch with you, but this is a problem that we come across specifically on MacOS. It has to do with linking of dynamic libraries with the executable when the model is called. 

I currently have a version of GOTMr which is working on [GitHub Actions ](https://github.com/aemon-j/GOTMr/actions/runs/880068285) so you could try to install that one:

`devtools::install_github("aemon-j/GOTMr", ref = "lake")`

But you will also need to install a couple of dependencies using homebrew from the terminal:

```
brew install gcc
brew install netcdf
```

Let me know if this works for you.

**Comment by rotifer16** [comment]
Dear all,
with these new instructions all models except for Simstrat work; for me
this is fine.
Thank you so much for your help
all the best,
Ulrike

Am Do., 12. Aug. 2021 um 20:10 Uhr schrieb Tadhg Moore <
***@***.***>:

**Comment by gbastienbeaudet** [comment]
Hi,
Do we have news on this issue ?
I have been having the same issue since installing the newer version of R (4.1.1). I am only using Simstrat. Simulations that were running smoothly before the updates are now giving the following output when using the command "run_ensemble" :

> run_ensemble(config_file = config_file, model = model)
Loading water temperature observations...[2022-01-11 22:13:35]
Finished loading water temperature observations![2022-01-11 22:13:35]
Running models... (Have you tried parallelizing?) [2022-01-11 22:13:35]
sh: line 1:  3813 Abort trap: 6           '/Library/Frameworks/R.framework/Versions/4.1/Resources/library/SimstratR/exec/simstrat' simstrat.par > /dev/null 2> /dev/null
Simstrat run is complete! [2022-01-11 22:13:35]
Error in file(file, "rt") : cannot open the connection
In addition: Warning message:
In file(file, "rt") :
  cannot open file './Simstrat/output/T_out.dat': No such file or directory

Best,
Gab

**Comment by robertladwig** [comment]
Hi Gab, 

I have uploaded a new Simstrat v3.0 binary for macOS to my repository. Could you please delete your old SimstratR version 
`remove.packages("SimstratR")`
and install this one 
`devtools::install_github("robertladwig/SimstratR", ref = "macOS)`
and see if LakeEnsemblR works now for you?

Thanks!
Robert

**Comment by gbastienbeaudet** [comment]
Hello Robert,

For some reasons, the installation line you provided didn't work, so I did the installation that way instead:

devtools::install_github("aemon-j/SimstratR")

But I am still getting the same error messages.

Best,
Gab

**Comment by gbastienbeaudet** [comment]
Hello Robert,

After the installation of your version, I am still getting error message but a little bit shorter this time:

> # 2. Run ensemble lake models
> run_ensemble(config_file = config_file, model = model)
Loading water temperature observations...[2022-01-12 19:04:32]
Finished loading water temperature observations![2022-01-12 19:04:32]
Running models... (Have you tried parallelizing?) [2022-01-12 19:04:33]
Simstrat run is complete! [2022-01-12 19:04:33]
Error in file(file, "rt") : cannot open the connection
In addition: Warning message:
In file(file, "rt") :
  cannot open file './Simstrat/output/T_out.dat': No such file or directory

Do you think of anything that might be causing this ?

Cheers,
Gab

**Comment by robertladwig** [comment]
Could you run the provided Simstrat executable from SimstratR (SimstratR/exec/simstrat) inside your Simstrat files please?
`./../SimstratR/exec/simstrat`
This would give us more information and an error message to see why Simstrat did not produce any results.

**Comment by gbastienbeaudet** [comment]
Of course. This is what I am getting.

 Simstrat version 3.0
 Coupled with the biogeochemical library AED2
 This software has been developed at Eawag - Swiss Federal Institute of Aquatic Science and Technology
 
 [ERROR] File simstrat.par does not exist
logout

Saving session...
...copying shared history...
...saving history...truncating history files...
...completed.
Deleting expired sessions...none found.

The simstrat.par file does however exist, which is strange.

**Comment by robertladwig** [comment]
Were you in the right folder (the one where you have simstrat.par)? I did test it on my Mac, and for Simstrat v3.0 I get this error 
```
  "ModelConfig" : {
    "MaxLengthInputData"      : 1000,                   ! Maximum length of input data (initial conditions, bathymetry, grid etc.)
    "CoupleAED2"              : false,                  ! Switch to turn on/off the biochemical model AED2
    "TurbulenceModel"         : 1,                      ! 1: k-epsilon, 2: Mellor-Yamada
    "SplitSeicheParameter"    : false,                  ! True: Use a_seiche if N2 exceeds strat_sumr and a_seiche_w otherwise, False: Always use a_seiche
    "StabilityFunction"       : 2,                      ! 1: constant, 2: quasi-equilibrium
    "FluxCondition"           : 1,                      ! 0: Dirichlet condition, 1: no-flux
    "Forcing"                 : 5,! /LER/ 1: Wind + Temp + SolRad, 2: 1 + VapP, 3: 2 + Cloud, 4: Wind + HeatFlux + SolRad, 5: 2 + Incoming Long Wave
    "UserDefinedWaterAlbedo"  : false,                  ! True: Computed monthly water albedo values from Grishchenko are used, False: Water albedo is user defined (below) and constant
    "UseFilteredWind"         : false,                  ! Switch to activate filtered wind (needs one more column in forcing file)
    "SeicheNormalization"     : 2,                      ! 1: max N2, 2: integral
    "WindDragModel"           : 3,                      ! 1: constant, 2: ocean (increasing), 3: lake (Wüest and Lorke, 2003)
    "InflowPlacement"         : 1,                      ! 0: inflow depths are chosen manually, 1: inflow is density-driven
    "PressureGradients"       : 0,                      ! 0: off, 1: Svensson 1978
    "IceModel"                : 1,! /LER/ Switch to turn on/off the ice model
    "SnowModel"               : 1,! /LER/ Switch to turn on/off the snow model (needs ice model and an additional column in forcing)
    "InflowMode"              : 1,
```

Could you try adding  "InflowMode"              : 1, to your simstrat.par file and run it again please?

**Comment by gbastienbeaudet** [comment]
I was indeed in the good folder, and the addition of InflowMode did not change the error.

**Comment by robertladwig** [comment]
Just to be sure, if you start a terminal and run 
`pwd`
you see the simstrat.par, and then running 
`./../SimstratR/exec/simstrat` (to wherever SimstratR is located on your Mac)
gives you still that error "[ERROR] File simstrat.par does not exist"?

**Comment by gbastienbeaudet** [comment]
When going throught the terminal, I am getting the following now:

Simstrat version 3.0
 Coupled with the biogeochemical library AED2
 This software has been developed at Eawag - Swiss Federal Institute of Aquatic Science and Technology
 
 [OK] Configuration: simstrat.par
 [OK] Grid file successfully read
dyld[10815]: missing symbol called

Program received signal SIGABRT: Process abort signal.

Backtrace for this error:
#0  0x10455e33d
#1  0x10455d76d
#2  0x7ff803b4de2c
Abort trap: 6
Gabriels-MacBook-Air:Simstrat smokeyhandz$

**Comment by robertladwig** [comment]
Okay that dyld error will make it hard to fix it. Which macOS version are you running?

And if you have "otool" installed, could you please post the output of the /SimstratR/exec/simstrat file
`otool -L simstrat`

Mine reads 
```
(base) Roberts-iMac:exec robertladwig$ otool -L simstrat 
simstrat:
	@executable_path/libgfortran.5.dylib (compatibility version 6.0.0, current version 6.0.0)
	/usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 1281.100.1)
	@executable_path/libgcc_s.1.dylib (compatibility version 1.0.0, current version 1.0.0)
	@executable_path/libquadmath.0.dylib (compatibility version 1.0.0, current version 1.0.0)
```

**Comment by gbastienbeaudet** [comment]
I am on macOS Monterey (12.0.1).

I get the same output as yours:

simstrat:
	@executable_path/libgfortran.5.dylib (compatibility version 6.0.0, current version 6.0.0)
	/usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 1281.100.1)
	@executable_path/libgcc_s.1.dylib (compatibility version 1.0.0, current version 1.0.0)
	@executable_path/libquadmath.0.dylib (compatibility version 1.0.0, current version 1.0.0)

**Comment by robertladwig** [comment]
Could you please test if you can run LakeEnsemblR on the Feeagh example (https://github.com/aemon-j/LakeEnsemblR/tree/main/inst/extdata/feeagh) using Simstrat?

**Comment by gbastienbeaudet** [comment]
I can't. Getting the same errors.

**Comment by gbastienbeaudet** [comment]
Good morning Robert,

This morning I played with different simulations to try and follow up on everything we talked yesterday. For some reasons, I was able to make everything work ! In the end, I am not sure what was causing the initial errors. I had to play a little bit with different parameters until it magically worked. When putting every parameters back to their original status (where it was crashing yesterday), it worked.

I think their might have been changes in the YAML file layout that was causing some problems. Also, relating to an issue we had yesterday, I noticed with your version of SimstratR that "export_config" is generating a "simstrat.par" with "InflowPlacement" instead of "InflowMode".

Anyway, thank you so much for your help, it was greatly appreciated !!! Cheers !!
Gab
